test-metadata-generation.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+source tools.sh
+source createdisk-library.sh
+source snc-library.sh
+
+# If the user set OKD_VERSION in the environment, then use it to override OPENSHIFT_VERSION, set BASE_OS, and set USE_LUKS
+# Unless, those variables are explicitly set as well.
+OKD_VERSION=${OKD_VERSION:-none}
+if [[ ${OKD_VERSION} != "none" ]]
+then
+    OPENSHIFT_VERSION=${OKD_VERSION}
+    BASE_OS=fedora-coreos
+fi
+BASE_OS=${BASE_OS:-rhcos}
+USE_LUKS=${USE_LUKS:-false}
+
+# If the user set OKD_VERSION in the environment, then use it to override OPENSHIFT_VERSION, MIRROR, and OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+# Unless, those variables are explicitly set as well.
+OKD_VERSION=${OKD_VERSION:-none}
+if [[ ${OKD_VERSION} != "none" ]]
+then
+    OPENSHIFT_VERSION=${OKD_VERSION}
+    MIRROR=${MIRROR:-https://github.com/openshift/okd/releases/download}
+fi
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+OPENSHIFT_RELEASE_VERSION="${OPENSHIFT_VERSION-4.7.0}"
+
+# CRC_VM_NAME: short VM name to use in crc_libvirt.sh
+# BASE_DOMAIN: domain used for the cluster
+# VM_PREFIX: full VM name with the random string generated by openshift-installer
+CRC_VM_NAME=${CRC_VM_NAME:-crc}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+
+VM_PREFIX=${CRC_VM_NAME}-abcde
+INTERNAL_IP=192.168.126.11
+
+CRC_ZSTD_EXTRA_FLAGS="--fast"
+
+# Prepare fake directory structure matching create_disk expectations
+baseDir="test-metadata-generation"
+mkdir $baseDir
+cd $baseDir
+srcDir=src
+destDir=dest
+
+mkdir -p "$srcDir"
+mkdir -p "$srcDir/auth"
+touch "$srcDir"/auth/kubeconfig
+touch "$srcDir"/auth/kubeadmin-password
+touch id_ecdsa_crc
+touch "$srcDir"/vmlinuz-0.0.0
+touch "$srcDir"/initramfs-0.0.0.img
+
+echo {} | ${JQ} '.version = "1.1"' \
+    | ${JQ} '.type = "snc"' \
+    | ${JQ} ".buildInfo.buildTime = \"$(date -u --iso-8601=seconds)\"" \
+    | ${JQ} ".buildInfo.openshiftInstallerVersion = \"0.0.0\"" \
+    | ${JQ} ".buildInfo.sncVersion = \"xxx\"" \
+    | ${JQ} ".clusterInfo.openshiftVersion = \"${OPENSHIFT_RELEASE_VERSION}\"" \
+    | ${JQ} ".clusterInfo.clusterName = \"${CRC_VM_NAME}\"" \
+    | ${JQ} ".clusterInfo.baseDomain = \"${BASE_DOMAIN}\"" \
+    | ${JQ} ".clusterInfo.appsDomain = \"apps-${CRC_VM_NAME}.${BASE_DOMAIN}\"" >${srcDir}/crc-bundle-info.json
+
+mkdir -p "$destDir/linux"
+#mkdir -p "$destDir/macos"
+#mkdir -p "$destDir/windows"
```

Comment:
Should this be un-commented ?

Reply:
Ah no, it should be removed, `generate_{hyperkit,hyperv}_bundle` start with `mkdir "$destDir"`